compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.desktop.kt

```diff
@@ -265,29 +291,55 @@ internal class ComposeSceneMediator(
         component.addInputMethodListener(inputMethodListener)
         component.addFocusListener(focusListener)
         component.addMouseListener(mouseListener)
-        component.addMouseMotionListener(mouseMotionListener)
-        component.addMouseWheelListener(mouseWheelListener)
+        component.addMouseMotionListener(mouseListener)
+        component.addMouseWheelListener(mouseListener)
         component.addKeyListener(keyListener)
     }
 
     private fun unsubscribe(component: Component) {
         component.removeInputMethodListener(inputMethodListener)
         component.removeFocusListener(focusListener)
         component.removeMouseListener(mouseListener)
-        component.removeMouseMotionListener(mouseMotionListener)
-        component.removeMouseWheelListener(mouseWheelListener)
+        component.removeMouseMotionListener(mouseListener)
+        component.removeMouseWheelListener(mouseListener)
         component.removeKeyListener(keyListener)
     }
 
+    private var isMouseEventProcessing = false
+    private var lastMouseEvent = WeakReference<MouseEvent?>(null)
+    private inline fun processMouseEvent(event: MouseEvent, block: () -> Unit) {
+        // Track if [event] is currently processing to avoid recursion in case if [SwingPanel]
+        // manually spawns a new AWT event for interop view.
+        // See [InteropPointerInputModifier] for details.
+        isMouseEventProcessing = true
+        // Remember [event] itself as precaution of changes inside JDK if the same event is sent
+        // to multiple components inside [container]. In case of such changes it might lead to
+        // double processing of the same event because our mouse listener is added into several
+        // components.
+        lastMouseEvent = WeakReference(event)
+        try {
+            block()
+        } finally {
+            isMouseEventProcessing = false
+        }
+    }
+
     // Decides which AWT events should be delivered, and which should be filtered out
     private val awtEventFilter = object {
 
         var isPrimaryButtonPressed = false
 
         fun shouldSendMouseEvent(event: MouseEvent): Boolean {
             // AWT can send events after the window is disposed
-            if (isDisposed)
+            if (isDisposed) {
                 return false
+            }
+
+            // Filter out mouse event if [ComposeScene] is already processing this mouse event,
+            // or it was already received via another listener.
+            if (isMouseEventProcessing || event == lastMouseEvent.get()) {
```

Comment:
It's better to put this filtering closer to its context, in processMouseEvent.

`awtEventFilter` is about filtering events AWT sends that we don't want, not events we ourselves send.

Reply:
Done